README
This archive contains the distribution CPANPLUS-Dist-Slackware.

CPANPLUS-Dist-Slackware is a CPANPLUS plugin that creates Slackware compatible
packages from Perl distributions.  You can either install the created packages
using the API provided by CPANPLUS or manually with "installpkg".

    $ cpanp
    CPAN Terminal> i Some::Module --format=CPANPLUS::Dist::Slackware

    $ cpan2dist --format CPANPLUS::Dist::Slackware Some::Module
    $ sudo /sbin/installpkg /tmp/perl-Some-Module-1.0-i486-1_CPANPLUS.tgz

INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

DEPENDENCIES

Slackware Linux 13.37 and Perl 5.12.3 or better and the sudo command are
required.  The fakeroot command is highly recommended.  You can download a
script that builds fakeroot from http://slackbuilds.org/.

SUPPORT AND DOCUMENTATION

Type "perldoc CPANPLUS::Dist::Slackware" after installation to see the module
usage information.

If you want to hack on the source it might be a good idea to install
Dist::Zilla and grab the latest version with git using the command:

    git clone https://github.com/voegelas/CPANPLUS-Dist-Slackware.git

LICENSE AND COPYRIGHT

Copyright 2012-2016 Andreas Voegele

This library is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

See http://dev.perl.org/licenses/ for more information.



Changes
Revision history for CPANPLUS-Dist-Slackware

1.026  2018-03-30 08:24:28 CEST

  - Support site installations in /usr/local. Example:
    cpanp i Mojolicious --format=CPANPLUS::Dist::Slackware --installdirs=site
  - Fix console output.

1.025  2017-10-08 18:54:06 CEST

  - Support prefixes other than /usr, e.g. /opt/cperl.
  - Depend on CPANPLUS 0.9166, which provides Autoflush.
  - Update the Net::Pcap plugin.
  - Remove the Crypt::SSLeay and GD plugins.
  - Change into the working directory before the plugins are called.
  - Move some functions to the new Util module.
  - Restore compatibility with Perl 5.6.1 for testing purposes.
    The lowest supported Module::Build version is 0.36.

1.024  2016-09-11 23:37:16 CEST

  - The package architecture now defaults to "i586" on x86-based platforms.

1.023  2015-11-01 10:13:52 CET

  - Add a Search::Xapian plugin that fixes the distributions's Makefile.PL
    script.
  - Update the documentation.

1.022  2015-08-18 08:41:04 CEST

  - Escape a left brace in a regex to get rid of a deprecation warning.

1.021  2015-01-18 09:16:51 CET

  - Remove the YAML::LibYAML plugin as the RT tickets #46507 and #74238 have
    been solved.
  - Check whether a dependency is still distributed with Perl.
  - Depend on Perl 5.12.3.

1.020  2014-12-30 13:26:13 CET

  - Revert to Dist::Zilla.

1.019  2014-12-30 08:31:51 CET

  - Avoid a warning in Module::CoreList.
  - Keep the .packlist files.
  - Depend on Perl 5.12.
  - Build the distribution with Dist::Milla.
  - Add a cpanfile.

1.018  2014-12-05 18:03:07 CET

  - Adapt the GD plugin to GD 2.56.

1.017  2014-08-17 21:03:26 CEST

  - Add a GD plugin, which patches the module's tests.
  - Make Alien::wxWidgets use the system's wxWidgets library if the environment
    variable WX_CONFIG is set to the full path to wx-config.
  - Patch the Crypt::SSLeay distribution's Makefile.PL if necessary.
  - Remove the Padre plugin as the development seems to have stopped.
  - Update the list of modules that SpamAssassin can optionally utilize.
  - Add Dist::Zilla and Dist::Zilla::Test::Perl::Critic to dist.ini.

1.016  2013-12-21 19:22:16 CET

  - Use parent instead of base.

1.015  2013-10-13 10:55:57 CEST

  - Update the Net::Pcap plugin.
  - The Convert::BinHex plugin is no longer needed.

1.014  2013-09-11 12:12:12 CEST

  - List prerequisites that are distributed with Perl in README.SLACKWARE if
    the required version is higher than the supplied version.

1.013  2013-06-09 17:41:02 CEST

  - Remove Build.PL.

1.012  2013-06-09 12:07:57 CEST

  - Use Dist::Zilla to build the distribution.
  - Use _perl_mb_opt() in _fake_install() instead of hardcoded path names.

1.011  2013-05-30 10:56:48 CEST

  - Support Module::Build::Tiny by passing command-line options to 'Build
    install'.
  - Respect the environment variable BUILD.  As packages may be built
    recursively, setting the build number is mainly useful when all packages
    are rebuilt, e.g. after Perl has been upgraded.

1.010  2012-08-23 03:12:31 CEST

  - Use three digit version numbers.  Use the same version number everywhere.
  - The Padre plugin creates a doinst.sh file that runs update-desktop-database
    and gtk-update-icon-cache.
  - The YAML::Syck plugin is no longer needed.
  - Use the CPAN mail address as the contact address.

1.00  2012-02-13 09:48:12 CET

  - List required packages instead of module names in README.SLACKWARE.
  - Update the YAML::LibYAML plugin.  Add a YAML::Syck plugin.  Fix errors that
    occur when these modules are built with the GCC compiler flag
    "-Werror=format-security".
  - Add a Convert::BinHex plugin, which fixes a bug in the module's test suite.
  - Add a Net::Pcap plugin, which fixes the module's Makefile.PL.

0.05  2012-02-01 19:31:33 CET

  - Document the plugin interface.
  - Add a Padre plugin, which installs a desktop entry and an icon for Padre.
  - Add a Mail::SpamAssassin plugin, which installs an init script and additional
    documentation.
  - Don't prepend "perl-" to module names like "perl-ldap" that already start
    with "perl-".

0.04  2012-01-24 07:25:41 CET

  - Add plugins, which may be used to fix quirks or customize packages.
  - Make sure that makepkg can access the staging directory if fakeroot is not
    available.

0.03  2012-01-18 14:51:26 CET

  - Make sure that the staging directory's parent directory exists.

0.02  2012-01-18 11:13:00 CET

  - Warn if files are installed in non-standard directories.
  - Unless --keepsource is given, the staging directory is removed after a
    package has been built successfully.
  - Link to https://metacpan.org/release instead of /module.
  - Mention cpan2tgz, which I wasn't aware of.

0.01  2012-01-12 18:58:56 CET

  - First version, released on an unsuspecting world.



Build.PL
use 5.010;
use strict;
use warnings;
use Module::Build;

my $class = Module::Build->subclass(
    class => "Module::Build::Custom",
    code => <<'SUBCLASS' );

sub ACTION_critique {
    my $self = shift;
    $self->depends_on("test");
    $self->do_system(qw(perlcritic lib/));
}
SUBCLASS

my $builder = $class->new(
    module_name         => 'CPANPLUS::Dist::Slackware',
    license             => 'perl',
    dist_author         => q{Andreas Voegele <voegelas@cpan.org>},
    dist_version_from   => 'lib/CPANPLUS/Dist/Slackware.pm',
    build_requires => {
        'Module::Build' => 0.2808,
        'Test::More' => 0,
    },
    configure_requires => {
        'Module::Build' => 0.2808,
    },
    requires => {
        'perl' => 5.010,
        'CPANPLUS' => 0,
        'Cwd' => 0,
        'File::Find' => 0,
        'File::Spec' => 0,
        'File::Temp' => 0,
        'IO::Compress::Gzip' => 0,
        'IPC::Cmd' => 0,
        'Locale::Maketext::Simple' => 0,
        'Module::CoreList' => 0,
        'Module::Pluggable' => 0,
        'Params::Check' => 0,
        'Pod::Find' => 0,
        'Pod::Simple' => 0,
        'POSIX' => 0,
        'Text::Wrap' => 0,
        'version' => 0.77,
    },
    add_to_cleanup      => [ 'CPANPLUS-Dist-Slackware-*' ],
    create_makefile_pl => 'traditional',
    meta_merge     => {
        resources => {
            repository => 'git://github.com/voegelas/CPANPLUS-Dist-Slackware.git'
        },
    },
);

$builder->create_build_script();


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.011.
use strict;
use warnings;

use 5.006001;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Install Perl distributions on Slackware Linux",
  "AUTHOR" => "Andreas Voegele <voegelas\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "CPANPLUS-Dist-Slackware",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006001",
  "NAME" => "CPANPLUS::Dist::Slackware",
  "PREREQ_PM" => {
    "CPANPLUS" => "0.9166",
    "Config" => 0,
    "Cwd" => 0,
    "ExtUtils::Install" => 0,
    "File::Find" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Compress::Gzip" => 0,
    "IPC::Cmd" => 0,
    "Locale::Maketext::Simple" => 0,
    "Module::CoreList" => "2.32",
    "Module::Pluggable" => 0,
    "POSIX" => 0,
    "Params::Check" => 0,
    "Pod::Find" => 0,
    "Pod::Simple" => 0,
    "Text::Wrap" => 0,
    "base" => 0,
    "version" => "0.77"
  },
  "TEST_REQUIRES" => {
    "Test::More" => "0.96"
  },
  "VERSION" => "1.028",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPANPLUS" => "0.9166",
  "Config" => 0,
  "Cwd" => 0,
  "ExtUtils::Install" => 0,
  "File::Find" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Compress::Gzip" => 0,
  "IPC::Cmd" => 0,
  "Locale::Maketext::Simple" => 0,
  "Module::CoreList" => "2.32",
  "Module::Pluggable" => 0,
  "POSIX" => 0,
  "Params::Check" => 0,
  "Pod::Find" => 0,
  "Pod::Simple" => 0,
  "Test::More" => "0.96",
  "Text::Wrap" => 0,
  "base" => 0,
  "version" => "0.77"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
